app/assets/javascripts/discourse/app/routes/user-activity-pending.js

```diff
@@ -1,6 +1,13 @@
-import UserAction from "discourse/models/user-action";
-import UserActivityStreamRoute from "discourse/routes/user-activity-stream";
+import DiscourseRoute from "discourse/routes/discourse";
+import { observes } from "discourse-common/utils/decorators";
 
-export default UserActivityStreamRoute.extend({
-  userActionType: UserAction.TYPES.pending,
+export default DiscourseRoute.extend({
+  model() {
+    return this.store.findAll("pending-post");
+  },
+
+  @observes("currentUser.pending_posts_count")
```

Comment:
We are trying to move away from observers. Is there any app event we can subscribe to when the pending posts come in?

Reply:
> When speaking of app event, do you have something else in mind than the `MessageBus` event?

Yup, look for `<model>.appEvents.trigger()` calls.

Reply:
Thanks, will do that! :ok_hand:

Reply:
The code I just pushed is now using `appEvents` as suggested :slightly_smiling_face: As I didn’t find any useful event for my case, I added `current-user:updated` where we call `currentUser.setProperties` in response to a `MessageBus` event.